modules/dnn/misc/python/test/test_dnn.py

```diff
@@ -286,38 +286,40 @@ def test_textdetection_model(self):
 
 
     def test_face_detection(self):
-        proto = self.find_dnn_file('dnn/opencv_face_detector.prototxt')
-        model = self.find_dnn_file('dnn/opencv_face_detector.caffemodel', required=False)
-        if proto is None or model is None:
-            raise unittest.SkipTest("Missing DNN test files (dnn/opencv_face_detector.{prototxt/caffemodel}). Verify OPENCV_DNN_TEST_DATA_PATH configuration parameter.")
-
+        model = self.find_dnn_file('dnn/onnx/models/yunet-202303.onnx', required=False)
         img = self.get_sample('gpu/lbpcascade/er.png')
-        blob = cv.dnn.blobFromImage(img, mean=(104, 177, 123), swapRB=False, crop=False)
 
-        ref = [[0, 1, 0.99520785, 0.80997437, 0.16379407, 0.87996572, 0.26685631],
-               [0, 1, 0.9934696,  0.2831718,  0.50738752, 0.345781,   0.5985168],
-               [0, 1, 0.99096733, 0.13629119, 0.24892329, 0.19756334, 0.3310290],
-               [0, 1, 0.98977017, 0.23901358, 0.09084064, 0.29902688, 0.1769477],
-               [0, 1, 0.97203469, 0.67965847, 0.06876482, 0.73999709, 0.1513494],
-               [0, 1, 0.95097077, 0.51901293, 0.45863652, 0.5777427,  0.5347801]]
+        ref = [[1, 339.62445, 35.32416, 30.754604, 40.202126, 0.9302596],
+               [1, 140.63962, 255.55545, 32.832615, 41.767395, 0.916015],
+               [1, 68.39314, 126.74046, 30.29324, 39.14823, 0.90639645],
+               [1, 119.57139, 48.482178, 30.600697, 40.485996, 0.906021],
+               [1, 259.0921, 229.30713, 31.088186, 39.74022, 0.90490955],
+               [1, 405.69778, 87.28158, 33.393406, 42.96226, 0.8996978]]
+
+        net = cv.FaceDetectorYN.create(
+            model=model,
+            config="",
+            input_size=img.shape[:2],
+            score_threshold=0.3,
+            nms_threshold=0.45,
+            top_k=5000,
+        )
 
         print('\n')
         for backend, target in self.dnnBackendsAndTargets:
```

Comment:
backend and target is not set in the loop. All iterations use default back-end. FaceDetectorYN has backend and target options in factory method. I propose to recreate the object on each iteration with properly set backend parameters.

Reply:
Yes. FaceDetectorYN doesn't export some methods like setBackend and setTarget. I will make it create a new object for each BackendAndTargets